src/algokit/cli/deploy.py

```diff
@@ -0,0 +1,104 @@
+import logging
+import os
+import shlex
+import typing as t
+from pathlib import Path
+
+import click
+
+from algokit.core import proc
+from algokit.core.conf import ALGOKIT_CONFIG
+from algokit.core.deploy import load_deploy_config, load_env_files
+
+logger = logging.getLogger(__name__)
+
+
+def _ensure_environment_secrets(
+    config_env: dict[str, str], environment_secrets: list[str], *, skip_mnemonics_prompts: bool
+) -> None:
+    for key in environment_secrets:
+        if not config_env.get(key):
+            if skip_mnemonics_prompts:
+                raise click.ClickException(f"Error: missing {key} environment variable")
+            config_env[key] = click.prompt(key, hide_input=True)
+
+
+class CommandParamType(click.types.StringParamType):
+    name = "command"
+
+    def convert(
+        self, value: t.Any, param: click.Parameter | None, ctx: click.Context | None  # noqa: ANN401
+    ) -> list[str]:
+        str_value = super().convert(value=value, param=param, ctx=ctx)
+        try:
+            return shlex.split(str_value)
+        except Exception as ex:
+            logger.debug(f"Failed to parse command string: {str_value}", exc_info=True)
+            raise click.BadParameter(str(ex), param=param, ctx=ctx) from ex
+
+
+@click.command("deploy")
+@click.argument("environment_name", default=None, required=False)
+@click.option(
+    "--command",
+    "-C",
+    type=CommandParamType(),
+    default=None,
+    help="Custom deploy command. If not provided, will load the deploy command from .algokit.toml file.",
+)
+@click.option(
+    "--interactive/--non-interactive",
+    " /--ci",  # this aliases --non-interactive to --ci
+    default=lambda: "CI" not in os.environ,
+    help="Enable/disable interactive prompts. If the CI environment variable is set, defaults to non-interactive",
+)
+@click.option(
+    "--path",
+    "-P",
+    type=click.Path(exists=True, readable=True, file_okay=False, resolve_path=True, path_type=Path),
+    default=".",
+    help="Specify the project directory. If not provided, current working directory will be used.",
+)
+def deploy_command(
+    *,
+    environment_name: str | None,
+    command: list[str] | None,
+    interactive: bool,
+    path: Path,
+) -> None:
+    """Deploy smart contracts from AlgoKit compliant repository."""
+    if environment_name:
```

Comment:
if we lower case and the user has any uppercase characters and is using a case-sensitive file system, this means they won't be able to use that file

Reply:
@achidlow refined docs to mention that point and removed case insensitivity